src/h5web/providers/jupyter/utils.test.ts

```diff
@@ -0,0 +1,89 @@
+import { HDF5TypeClass } from '../hdf5-models';
+import { convertDtype } from './utils';
+
+describe('JupyterProvider utilities', () => {
+  describe('convertDtype', () => {
+    it('should convert unicode string dtypes', () => {
+      expect(convertDtype('|U25')).toEqual({
+        class: HDF5TypeClass.String,
+        charSet: 'H5T_CSET_UTF8',
+        strPad: 'H5T_STR_NULLPAD',
+        length: 25,
+      });
+    });
+
+    it('should convert bytes string dtypes', () => {
+      expect(convertDtype('|S')).toEqual({
+        class: HDF5TypeClass.String,
+        charSet: 'H5T_CSET_ASCII',
```

Comment:
Should we store this more concisely like the endianness?

Reply:
I would need to change the parsing of string types in the HsdsProvider but I would agree.

Having a field called `charSet` with `CSET` in the value is quite redundant.